dependency_manager/src/edm_tool/edm.py

```diff
@@ -751,8 +838,309 @@ def create_vscode_workspace_file(workspace_path: Path, workspace_checkout: dict)
+            try:
+                workspace_config = yaml.safe_load(config_file)
+            except yaml.YAMLError as e:
+                log.error(f"Error parsing yaml of \"{config_file}\": {e}")
+
+        args.config = config_path
+
+    working_dir = Path(args.working_dir).expanduser().resolve()
+
+    if not args.workspace:
+        log.info(f"No workspace provided, using current working dir {working_dir}")
+        args.workspace = working_dir
+
+    EDM.setup_workspace_from_config(args.workspace, args.config, args.update, args.create_vscode_workspace)
+
+    # add workspace to list of workspaces
+    if not args.workspace_name:
+        workspace_name = args.workspace.name
+        log.info(f"No workspace name given, using parent directory name {workspace_name}")
+        args.workspace_name = workspace_name
+
+    log.info(f"Add workspace {args.workspace_name} to list of workspaces")
+
+    workspace_name = args.workspace_name.replace("[", "").replace("]", "")
+
+    # write config file
+    edm_config_dir_path.mkdir(parents=True, exist_ok=True)
+    config = load_edm_config()
+
+    if not config:
+        log.info("Config is None, creating new config")
+        config = {}
+        config["edm"] = {}  # for general workspace independent config settings
+        config["workspaces"] = {}
+
+    with open(edm_config_path, 'w', encoding='utf-8') as edm_config_file:
+        config["edm"]["active_workspace"] = workspace_name
+        config["workspaces"][workspace_name] = {}
+        config["workspaces"][workspace_name]["path"] = args.workspace.as_posix()
+        config["workspaces"][workspace_name]["config"] = workspace_config
+        yaml.dump(config, edm_config_file)
+        log.info(f"Successfully saved edm config \"{edm_config_path}\".")
+
+
+def list_handler(_args):
+    """Handler for the edm list subcommand"""
+    log.info("Listing workspaces")
+    config = load_edm_config()
+
+    if not config:
+        log.info("No edm config found")
+        sys.exit(0)
+
+    for workspace_name, workspace_config in config["workspaces"].items():
+        log.info(f"  {workspace_name} ({workspace_config['path']})")
+
+
+def cd_handler(args):
+    """Handler for the edm cd subcommand"""
+    log.info("Changing workspace")
+    config = load_edm_config()
+
+    if not config:
+        log.error("No edm config found")
+        sys.exit(0)
+
+    if not args.workspace_name:
+        # load last active workspace from config
+        args.workspace_name = config["edm"]["active_workspace"]
+        if not args.workspace_name:
+            log.error("No active workspace found, can't change to a workspace I don't know")
+            sys.exit(0)
+
+    if args.workspace_name not in config["workspaces"]:
+        log.error(f"Can't find workspace {args.workspace_name} in config")
+        sys.exit(0)
+
+    # TODO: change active_workspace in config before calling chdir
+    os.chdir(config["workspaces"][args.workspace_name]["path"])
```

Comment:
Does this work for the parent process?

Reply:
Oh that's actually just an artifact of something I wanted to complete remove at least from this version. There was actually an os.exec call just after this statement but that has already been removed. However I would lean towards just droping the whole edm cd command for now